packages/app-store/googlevideo/components/InstallAppButton.tsx

```diff
@@ -0,0 +1,71 @@
+import { useState, useEffect } from "react";
+
+import type { InstallAppButtonProps } from "@calcom/app-store/types";
+import useApp from "@calcom/lib/hooks/useApp";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import { trpc } from "@calcom/trpc/react";
+import type { DialogProps } from "@calcom/ui";
+import { Button } from "@calcom/ui";
+import { Dialog, DialogClose, DialogContent, DialogFooter } from "@calcom/ui";
+
+import useAddAppMutation from "../../_utils/useAddAppMutation";
+
+export default function InstallAppButton(props: InstallAppButtonProps) {
+  const [showWarningDialog, setShowWarningDialog] = useState(false);
+
+  console.log(useApp("google-calendar").data);
+
+  return (
+    <>
+      {props.render({
+        onClick() {
+          setShowWarningDialog(true);
+          // mutation.mutate("");
+        },
+        disabled: showWarningDialog,
+      })}
+      <WarningDialog open={showWarningDialog} onOpenChange={setShowWarningDialog} />
+    </>
+  );
+}
+
+function WarningDialog(props: DialogProps) {
+  const { t } = useLocale();
+  const [googleCalendarPresent, setGoogleCalendarPresent] = useState<undefined | boolean>(undefined);
+  const googleCalendarData = useApp("google-calendar");
+
+  useEffect(() => {
+    if (googleCalendarData.data) setGoogleCalendarPresent(googleCalendarData.data.isInstalled > 0);
+  }, [googleCalendarData]);
```

Comment:
Is this the best way to use the `useApp` hook in this context?

Reply:
Left a comment

Reply:
https://github.com/calcom/cal.com/pull/5904#discussion_r1063647087